docs/reference/reference-deploy/config/agent-config-reference.rst

```diff
@@ -93,3 +93,13 @@
 
 -  ``container_auto_remove_disabled`` (debug): Whether to disable setting ``AutoRemove`` flag on
    task containers. Defaults to false.
+
+-  ``hooks``: Configuration for commands to run when certain events occur. The value of each option
+   in this section is an array of strings specifying the command and its arguments.
+
+   -  ``on_connection_failure``: A command to run when the agent fails to either connect to the
```

Comment:
Can we document the timeout behavior here? i.e., as implemented the agent will try to reconnect multiple times before giving up and invoking this hook.

I wonder if the name for this behavior is ideal: `on_connection_failure` suggests to me that the hook is invoked whenever a connection attempt fails, rather than only once at the end of a sequence of connection attempts + backoffs. Not sure if I can think of a better name offhand, though...

Reply:
`on_connection_lost` sgtm!